### ISSUE TITLE
Support `Invoking` suspend function

### ISSUE THREAD

**Issue by ychescale9** [MarkusAmshove/Kluent#151]
Right now to assert that invoking a suspend function throws an exception we have to do something like:

```kotlin
@Test
fun testASuspendFunctionThatThrowsTheExpectedException() {
    suspend fun func() {
        suspendCancellableCoroutine<Any> { throw IndexOutOfBoundsException() }
    }
    invoking {
        runBlocking {
            func()
        }
    } shouldThrow IndexOutOfBoundsException::class
}
```

Coroutines's `runBlocking` and `runBlockingTest` (from `kotlinx-coroutines-test` artifact) functions already provide a CoroutineScope as a lambda so it would be good to have a version of `invoking` that works with suspend functions:

```kotlin
@Test
fun testASuspendFunctionThatThrowsTheExpectedException() = runBlockingTest {
    suspend fun func() {
        suspendCancellableCoroutine<Any> { throw IndexOutOfBoundsException() }
    }
    invokingSuspend {
        func()
    } shouldThrow IndexOutOfBoundsException::class
}
```

I have this working locally and would love to send a PR if this makes sense. 
Thanks!

**Comment by ychescale9** [comment]
[Implementation](https://github.com/ReactiveCircus/FlowBinding/blob/master/kluent-suspend/src/main/java/reactivecircus/kluentsuspend/SuspendFunctionExceptions.kt) and [tests](https://github.com/ReactiveCircus/FlowBinding/blob/master/kluent-suspend/src/test/java/reactivecircus/kluentsuspend/SuspendFunctionExceptionsTest.kt).

**Comment by MarkusAmshove** [comment]
Hi @ychescale9 ,

this looks like a nice addition :-)

Feel free to send a PR!

**Comment by MarkusAmshove** [comment]
This is now implemented and released in v1.54 :-)
